subsys/usb/usb_descriptor.c

```diff
@@ -281,6 +286,41 @@ static struct usb_cfg_data *usb_get_cfg_data(struct usb_if_descriptor *iface)
 	return NULL;
 }
 
+/*
+ * Default USB SN string descriptor is CONFIG_USB_DEVICE_SN, but sometimes
+ * we want use another string as SN descriptor such as the chip's unique
+ * ID. So user can implement this function return a string thats will be
+ * replaced the default SN.
+ * Please note that the new SN descriptor you changed must has same length
+ * as CONFIG_USB_DEVICE_SN.
+ */
+__weak u8_t *usb_ascii_sn_descriptor_board_fixup(void)
+{
+	return NULL;
+}
+
+static void
+usb_fix_ascii_sn_string_descriptor(struct usb_sn_descriptor *sn)
```

Comment:
It does not fix anything, it only updates the string,  usb_update_sn_string_descriptor would be more accurate.